ova/Libraries/elastic_functions.sh

```diff
@@ -0,0 +1,247 @@
+
+}
+
+install_elasticsearch(){
+
+    yum install elasticsearch-${ELK_VERSION} -y
+    systemctl daemon-reload
+    systemctl enable elasticsearch.service
+
+}
+install_filebeat_7(){
+
+    yum install filebeat-${ELK_VERSION} -y
+    cp -f ${config_files}/filebeat.yml ${etc_filebeat}/filebeat.yml
+
+    sed -i "s/YOUR_ELASTIC_SERVER_IP/localhost/" ${etc_filebeat}/filebeat.yml
+    chmod go+r ${etc_filebeat}/filebeat.yml
+    curl -so ${etc_filebeat}/wazuh-template.json https://raw.githubusercontent.com/wazuh/wazuh/v${WAZUH_VERSION}/extensions/elasticsearch/7.x/wazuh-template.json
+    chmod go+r ${etc_filebeat}/wazuh-template.json
+
+    if  [ ${ELK_MINOR} -eq 2 ]; then
+        curl -s https://s3-us-west-1.amazonaws.com/packages-dev.wazuh.com/utils/wazuh-filebeat-module.tar.gz | tar -xvz --no-same-owner -C /usr/share/filebeat/module --owner=0
+    fi
+
+    systemctl daemon-reload
+    systemctl enable filebeat.service
+    systemctl start filebeat.service
+}
+configure_elasticsearch(){
+
+    cp -f ${config_files}/elasticsearch.yml ${etc_elastic}/elasticsearch.yml
+}
+
+configure_jvm_elastic(){
+
+    cp -f ${config_files}/elasticsearch.jvm ${etc_elastic}/jvm.options
+}
+
+configure_RAM(){
+
+    ram_gb=$(free -g | awk '/^Mem:/{print $2}')
+    ram=$(( ${ram_gb} / 2 ))
+    if [ ${ram} -eq "0" ]; then ram=1; fi
+    sed -i "s/-Xms16g/-Xms${ram}g/" ${etc_elastic}/jvm.options
+    sed -i "s/-Xmx16g/-Xms${ram}g/" ${etc_elastic}/jvm.options
+}
+
+configure_limitMEMLOCK(){
+
+    mkdir -p /etc/systemd/system/elasticsearch.service.d/
+    cp -f ${config_files}/elasticsearch.conf /etc/systemd/system/elasticsearch.service.d/elasticsearch.conf
+
+    systemctl daemon-reload
+    systemctl start elasticsearch.service
+}
+
+install_kibana(){
+
+    yum install kibana-${ELK_VERSION} -y
+}
+
+configure_kibana(){
+
+    openssl req -x509 -batch -nodes -days 3650 -newkey rsa:2048 -keyout ${etc_kibana}/kibana.key -out ${etc_kibana}/kibana.cert
+
+    cp -f ${config_files}/kibana.yml  ${etc_kibana}/kibana.yml
+
+    # Allow Kibana to listen on port 443
+    setcap 'CAP_NET_BIND_SERVICE=+eip' ${usr_kibana}/node/bin/node
+
+    # Configuring Kibana default settings
+    cp -f ${config_files}/kibana  /etc/default/kibana
+}
+
+install_kibana_app(){
+    if [ "${STATUS_PACKAGES}" == "stable" ]; then
+        #Wazuh-app production repository
+        sudo -u kibana ${usr_kibana}/bin/kibana-plugin install https://packages.wazuh.com/wazuhapp/wazuhapp-${WAZUH_VERSION}_${ELK_VERSION}.zip
+    fi
+
+    if [ "${STATUS_PACKAGES}" == "unstable" ]; then
+        # Wazuh-app pre-release repository
+        sudo -u kibana NODE_OPTIONS="--max-old-space-size=3072" ${usr_kibana}/bin/kibana-plugin install https://packages-dev.wazuh.com/pre-release/app/kibana/wazuhapp-${WAZUH_VERSION}_${ELK_VERSION}.zip
+    fi
+
+    systemctl daemon-reload
+    systemctl enable kibana.service
+    systemctl start kibana
+
+    while true; do
```

Comment:
This loop may be simplified if you use an `until ...; do stuff; done` structure. Here you can see an example: https://github.com/wazuh/wazuh-packages/blob/df2320802601b207c199c01425748724cca9eb65/ova/Libraries/elastic_functions.sh#L163-L166

Reply:
Done in https://github.com/wazuh/wazuh-packages/pull/223/commits/63e8681803657ce52428b8a65324cae5a2cc696c